spring-content-rest/src/main/java/internal/org/springframework/content/rest/utils/ContentStoreUtils.java

```diff
@@ -1,83 +1,146 @@
 import org.springframework.content.commons.annotations.OriginalFileName;
+import org.springframework.content.commons.io.MedializedResource;
 import org.springframework.content.commons.renditions.Renderable;
 import org.springframework.content.commons.repository.ContentStore;
 import org.springframework.content.commons.repository.Store;
 import org.springframework.content.commons.storeservice.ContentStoreInfo;
 import org.springframework.content.commons.storeservice.ContentStoreService;
 import org.springframework.content.commons.utils.BeanUtils;
-import org.springframework.content.rest.StoreRestResource;
 import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.core.io.Resource;
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.MediaType;
 import org.springframework.util.StringUtils;
 
+import internal.org.springframework.content.rest.StoreRestResource;
 import internal.org.springframework.content.rest.annotations.ContentStoreRestResource;
 
 public final class ContentStoreUtils {
 
-	private ContentStoreUtils() {}
-	
+	private ContentStoreUtils() {
+	}
+
 	/**
-	 * Given a store and a collection of mime types this method will iterate the mime-types returning the
-	 * first input stream that it can find from the store itself or, if the store implements Renderable
-	 * from a rendition.
+	 * Given a store and a collection of mime types this method will iterate the
+	 * mime-types returning the first input stream that it can find from the store
+	 * itself or, if the store implements Renderable from a rendition.
 	 * 
-	 * @param store		store the store to fetch the content from
-	 * @param mimeTypes	the mime types requested
-	 * @param entity 	the entity whose content is being fetched
-	 * @param headers 	headers that will be sent back to the client
+	 * @param store
+	 *            store the store to fetch the content from
+	 * @param mimeTypes
+	 *            the mime types requested
+	 * @param entity
+	 *            the entity whose content is being fetched
+	 * @param headers
+	 *            headers that will be sent back to the client
 	 * 
 	 * @return input stream
+	 * @throws IOException
 	 */
 	@SuppressWarnings("unchecked")
-	public static InputStream getContent(ContentStore<Object,Serializable> store, Object entity, List<MediaType> mimeTypes, HttpHeaders headers) {
+	public static InputStream getContent(ContentStore<Object, Serializable> store, Object entity,
+			List<MediaType> mimeTypes, HttpHeaders headers) {
 		InputStream content = null;
-		
-		Object entityMimeType = BeanUtils.getFieldWithAnnotation(entity, org.springframework.content.commons.annotations.MimeType.class);
+		String cName = "content";
+		String httpNameHeader = "x-file-name";
+
+		Object entityMimeType = BeanUtils.getFieldWithAnnotation(entity,
+				org.springframework.content.commons.annotations.MimeType.class);
 		if (entityMimeType == null)
 			return content;
-		
+
+		// content name
```

Comment:
What is this x-file-name header?  A quick browse on the internet indicated this is more about file upload that download?  Do we need this?

Reply:
Save from angular UI page. If response was loaded by ajax. It is necessary to have File name for save dialog. So I used it to have file name in response.

Reply:
OK.  Interesting.  I see.  Can your client-side not see the content-disposition header that we already add?  Are you running in a cross-origin setup here?  I wonder if perhaps we aren't exposing the content-disposition header as described in this stackoverflow (https://stackoverflow.com/a/41784348/5609837)?

Reply:
Omg! My bad. Yes, now I saw that there is part what will set "content-disposition" header. This is useless. We can let it be as You did using "content-disposition". Maybe  add this fact it in the examples. Cause it depends on "OriginalFileName" annotation. In the example this is not present (or I just not saw it :))  I probably miss it so when I did my proof of concept I just skip that part and did on my own way ( bad way ... ) I'll remove my superfluous code.